src/Eloquent/Builder.php

```diff
@@ -207,8 +214,14 @@ public function createOrFirst(array $attributes = [], array $values = []): Model
             try {
                 $document = $collection->findOneAndUpdate(
                     $attributes,
-                    ['$setOnInsert' => $values],
-                    ['upsert' => true, 'new' => true, 'typeMap' => ['root' => 'array', 'document' => 'array']],
+                    // Before MongoDB 5.0, the $setOnInsert does support empty document,
+                    // so the filter value are added to avoid an error.
```

Comment:
```suggestion
                    // Before MongoDB 5.0, $setOnInsert requires a non-empty document.
                    // This is should not be an issue as $values includes the query filter.
```

Optional re-wording.

Reply:
Thanks.